src/main/java/frc/robot/commands/SnapDrivebaseToAngle.java

```diff
@@ -11,59 +11,89 @@
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
 import frc.robot.subsystems.CommandSwerveDrivetrain;
+import frc.robot.subsystems.Vision.Vision;
+import frc.robot.subsystems.Vision.VisionIOLimelight;
+
 import java.util.Optional;
 
+import org.ejml.interfaces.decomposition.SingularValueDecomposition;
+
 /* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
 public class SnapDrivebaseToAngle extends Command {
     private CommandSwerveDrivetrain driveTrain;
     private CommandXboxController driverCont = new CommandXboxController(0);
     private double maxSpeed;
     private double angleToFace = 0.0;
+    private Vision vision;
+    private Vision aprilTagID;
 
     /** Creates a new SnapDrivebaseToAngle. */
-    public SnapDrivebaseToAngle(CommandSwerveDrivetrain driveTrain, double maxSpeed) {
+    public SnapDrivebaseToAngle(CommandSwerveDrivetrain driveTrain, double maxSpeed, Vision vision) {
         this.driveTrain = driveTrain;
         this.maxSpeed = maxSpeed;
+        this.vision = vision;
         // Use addRequirements() here to declare subsystem dependencies.
         addRequirements(driveTrain);
     }
 
     // Called when the command is initially scheduled.
     @Override
     public void initialize() {
-        if (driveTrain.getAngle() >= -30.0 && driveTrain.getAngle() <= 30.0) {
+        
+        // if (vision.getAprilTagID() == 18 || vision.getAprilTagID() == 7) {
```

Comment:
This is a killer if else statement. We can simplify it a fair bit by splitting these into 2 separate commends